### DISCUSSION TITLE
Unable to compile on go1.21

### DISCUSSION THREAD

**Discussion by rraj-splunk** [klauspost/compress#874] [Q&A]
Error faced:

```
external/com_github_klauspost_compress/zstd/dict.go:479:32: binary.LittleEndian.AppendUint32 undefined (type binary.littleEndian has no field or method AppendUint32)
external/com_github_klauspost_compress/zstd/dict.go:490:32: binary.LittleEndian.AppendUint32 undefined (type binary.littleEndian has no field or method AppendUint32)
external/com_github_klauspost_compress/zstd/dict.go:491:32: binary.LittleEndian.AppendUint32 undefined (type binary.littleEndian has no field or method AppendUint32)
external/com_github_klauspost_compress/zstd/dict.go:492:32: binary.LittleEndian.AppendUint32 undefined (type binary.littleEndian has no field or method AppendUint32)
compilepkg: error running subcommand external/go_sdk/pkg/tool/darwin_amd64/compile: exit status 2
```

Go version

```
$ go version
go version go1.21.0 darwin/amd64
```

Compress version:
```
github.com/klauspost/compress v1.17.0
```

**Comment by bartelsb** [comment]
This appears to still be an issue in `v1.17.1`.

**Reply by klauspost** [reply]
@bartelsb  I already answered this in #873

**Reply by bartelsb** [reply]
We're definitely using Go 1.21, but I think what's happening is that another dependency has a pinned dependency that references an older version of the `compress` package. Sorry, I'm not typically a Go developer so I may not be understanding how all this fits together correctly.

@rraj-splunk I used `go mod graph | grep compress` to find all references to the `compress` package. I found a dependency that specifies an old version of another dependency, and that old version uses `compress@v1.13.6`. That seems like the most likely culprit for me, and I'm guessing you have a similar situation too.

**Reply by klauspost** [reply]
@bartelsb That doesn't make sense, unless there is something broken with the package manager.

It will not use *both* the old and the new version.

**Reply by bartelsb** [reply]
You were right, I had one more file to update. We had a docker-compose file in a subdirectory that was getting used for tests, and I wasn't aware of it. That file still referenced Go 1.18. Sorry for the confusion, thanks for the help tracking it down!
